### DISCUSSION TITLE
vite-plugin-vanjs

### DISCUSSION THREAD

**Discussion by thednp** [vanjs-org/van#393] [Show and tell]
Hi @Tao-VanJS and everyone!

I've created the [vite-plugin-vanjs](https://github.com/thednp/vite-plugin-vanjs)  to work with [vanjs-core](https://github.com/vanjs-org/van), [van-ext](https://github.com/vanjs-org/van/tree/main/x) and [mini-van-plate](https://github.com/vanjs-org/mini-van) to fully enable SSR/SSG for your VanJS applications. You also have some vite starter templates to play with (check second reply).

**Features**:
* enables SSR for your apps auto-magically;
* enables JSX transformation

What do you guys think?

**Comment by thednp** [comment]
Published the plugin to [NPM](https://www.npmjs.com/package/vite-plugin-vanjs).

Please install a starter template `npm create vanjs@latest` and give it a try.

Powered by [vite-plugin-vanjs](https://github.com/thednp/vite-plugin-vanjs).

**Comment by thednp** [comment]
Update, now the plugin at v0.0.3 will automatically patch your application to work seamless with SSR/SSG.

@Tao-VanJS I still need your check on all this, when you have some time of course.

**Reply by Tao-VanJS** [reply]
I don't use vite too much. But if you're comfortable about your template, I can add it into the `Community Examples` section.

**Reply by thednp** [reply]
I know you're a Deno guy, probably some day we're all gonna be.

IMO VanJS is so good it should have a starter template along all others within the official vite starter templates, not just our Community Examples, which is why I think your input could be a bit more consistent.

**Comment by thednp** [comment]
@RobinManoli this plugin provides a router that was recently updated to work with hystory changes (back & forward buttons). It will soon also support file system routing.

**Comment by thednp** [comment]
**Our vite plugin is now async first!**

What this means is that the router now integrates better with data models via the `route.preload` and `route.load` params of Page components and orchestrates page hydration async **only**.

A much more simple model, more efficient and functional. Check the updated WIKIs and the SSR examples `create vanjs@latest`.

Have fun!
